pkg/gimletd/notifications/podStateManager.go

```diff
@@ -0,0 +1,131 @@
+package notifications
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/gimlet-io/gimlet-cli/pkg/dashboard/api"
+	"github.com/gimlet-io/gimlet-cli/pkg/dashboard/model"
+	"github.com/gimlet-io/gimlet-cli/pkg/dashboard/store"
+	"github.com/sirupsen/logrus"
+)
+
+type PodStateManager struct {
+	notifManager Manager
+	store        store.Store
+	waitTime     time.Duration
+}
+
+func NewPodStateManager(notifManager Manager, store store.Store, waitTime time.Duration) *PodStateManager {
+	return &PodStateManager{notifManager: notifManager, store: store, waitTime: waitTime}
+}
+
+func (p PodStateManager) Track(pods []*api.Pod) {
+	p.trackStates(pods)
+}
+
+func (p PodStateManager) Run() {
+	for {
+		p.setFiringState()
+
+		time.Sleep(p.waitTime * time.Minute)
+	}
+}
+
+func (p PodStateManager) trackStates(pods []*api.Pod) {
+	for _, pod := range pods {
+		podName := fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)
+		currentTime := time.Now().Unix()
+
+		if p.alreadyAlerted(podName) || p.statusNotChanged(podName, pod.Status) {
+			continue
+		}
+
+		if podErrorState(pod.Status) {
```

Comment:
The only difference in the two branches is the alertstate string.
Make this shorter, remove the repetition.